### PR TITLE
fix(mcp): enforce MAX_PAGE_SIZE limit on list tools to prevent oversized responses

### PR DESCRIPTION
**Pull request by aminghadersohi**
## Summary

- Add `MAX_PAGE_SIZE=100` and `DEFAULT_PAGE_SIZE=10` constants
- Add `le=MAX_PAGE_SIZE` constraint to `page_size` field in all three list request schemas (charts, dashboards, datasets) so Pydantic rejects values over 100
- Add defense-in-depth clamp in `ModelListCore.run_tool()` 
- Add missing middleware stack (response size guard, logging, error handler) to `__main__.py` entry point which had none
- Add tests for page_size boundary validation

### BEFORE/AFTER SCREENSHOTS OR ANIMATIONS

**Before:** LLMs could request `page_size=10000` and get back massive responses (e.g., 177k chars / ~50k tokens for 1000 charts), overwhelming client context windows and causing 5-10 minute delays.

**After:** `page_size` is capped at 100 via Pydantic schema validation. Values over 100 return a clear validation error:
```
Input should be less than or equal to 100
```

### TESTING INSTRUCTIONS

```python
# Should fail with validation error
list_charts(request={"page_size": 200})

# Should succeed
list_charts(request={"page_size": 100})
list_charts(request={"page_size": 10})
```

### ADDITIONAL INFORMATION

- `execute_sql` already has proper bounds (`le=10000`) — this brings list tools in line
- The `__main__.py` entry point (`python -m superset.mcp_service`) previously had zero middleware protection

___

## **CodeAnt-AI Description**
Limit list results and ensure MCP protections are active in every startup mode

### What Changed
- List charts, dashboards, and datasets now reject page sizes above 100, so requests can no longer return oversized result sets
- Requests at the limit still work, and invalid page sizes now fail with a clear validation error
- The MCP service now loads the standard safety, logging, and error-handling protections when started from the command line or other transport modes

### Impact
`✅ Smaller list responses`
`✅ Fewer slow or stalled requests`
`✅ Safer MCP startup across entry points`
<details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

File: tests/unit_tests/mcp_service/chart/tool/test_list_charts.py [hunk 1]
```diff
@@ -133,6 +134,19 @@ def test_invalid_page_number(self):
         with pytest.raises(ValueError, match="Input should be greater than 0"):
             ListChartsRequest(page_size=0)
 
+    def test_page_size_exceeds_max(self):
+        """Test that page_size over MAX_PAGE_SIZE raises validation error."""
+        with pytest.raises(
+            ValueError,
```
[the remark this answers is not in the document]
Comment (aminghadersohi) [reply tests/unit_tests/mcp_service/chart/tool/test_list_charts.py:140]: Thanks for the review! This is actually a false positive — in Pydantic V2, `ValidationError` is a subclass of `ValueError` (confirmed via `issubclass(pydantic.ValidationError, ValueError)` returning `True`), so `pytest.raises(ValueError)` correctly catches it. The existing tests in this same file (e.g., `test_invalid_page_number` at line 130) use the same pattern and pass in CI. No change needed here.

**Review by geido — APPROVED** [review]
